### ISSUE TITLE
hiera-eyaml failing to install with puppetserver_gem as a provider

### ISSUE THREAD

**Issue by jyaworski** [hunner/puppet-hiera#124]
I tried to install hiera-eyaml on puppetserver using puppetserver_gem, and got this error:

```puppet
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Duplicate declaration: Package[hiera-eyaml] is already declared in file /etc/puppet/environments/enable_eyaml/modules/hiera/manifests/install.pp:60; cannot redeclare at /etc/puppet/environments/enable_eyaml/modules/hiera/manifests/install.pp:65 on node puppet1.domain.com
```

The relevant code is here: https://github.com/hunner/puppet-hiera/blob/master/manifests/install.pp#L54-L69

I suspect there's a name collision.

**Comment by hunner** [comment]
#125

**Comment by jyaworski** [comment]
From IRC, just for the public.

Puppet < 4 uses title + name for uniqueness. Puppet 4 uses title + name + provider, making it more granular.

**Comment by jyaworski** [comment]
Closed via #125
